gammapy/analysis/steps.py

```diff
@@ -0,0 +1,498 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""classes containing the analysis steps supported by the high level interface"""
+
+import abc
+import os
+from astropy.coordinates import SkyCoord
+from astropy.table import Table
+from regions import CircleSkyRegion
+from gammapy.data import DataStore
+from gammapy.datasets import Datasets, FluxPointsDataset, MapDataset, SpectrumDataset
+from gammapy.estimators import (
+    ExcessMapEstimator,
+    FluxPointsEstimator,
+    LightCurveEstimator,
+)
+from gammapy.makers import (
+    FoVBackgroundMaker,
+    MapDatasetMaker,
+    ReflectedRegionsBackgroundMaker,
+    RingBackgroundMaker,
+    SafeMaskMaker,
+    SpectrumDatasetMaker,
+    DatasetsMaker
+)
+from gammapy.maps import Map, MapAxis, RegionGeom, WcsGeom
+from gammapy.utils.pbar import progress_bar
+from gammapy.utils.scripts import make_path, make_name
+
+
+class AnalysisStepBase(abc.ABC):
+    tag = "analysis-step"
+    requires_datasets = False
+    requires_models = False
+
+    def __init__(self, analysis, name=None, overwrite=True):
+        self.analysis = analysis
+        self.overwrite = overwrite
+        self._name = make_name(name)
```

Comment:
What is the name attribute used for? Is to generate the dataset later? But what happened for multiple datasets?